src/main/java/com/lucidworks/spark/SolrRelation.java

```diff
@@ -63,8 +74,17 @@ public SolrRelation(SQLContext sqlContext, Map<String,String> config, DataFrame
     solrQuery.set("collection", collection);
     if (dataFrame != null) {
       schema = dataFrame.schema();
-    } else {
-      schema = solrRDD.getQuerySchema(solrQuery);
+    }
+    else {
```

Comment:
nitpick, but go ahead and put the else on the same line as the bracket, like `} else {`.  Just to be consistent style-wise.